k8s/operators/src/pool/diskpool/client.rs

```diff
@@ -1,32 +1,61 @@
-use super::crd::{DiskPool, DiskPoolSpec};
+use super::{
+    crd::DiskPool,
+    new_crd::{DiskPool as NewDiskPool, DiskPoolSpec as NewDiskPoolSpec},
+};
 use crate::error::Error;
 use k8s_openapi::apiextensions_apiserver::pkg::apis::apiextensions::v1::CustomResourceDefinition;
-use kube::{api::PostParams, Api, Client, CustomResourceExt, ResourceExt};
-use openapi::apis::StatusCode;
-use tracing::info;
+use kube::{
+    api::{ListParams, Patch, PatchParams, PostParams},
+    core::crd::merge_crds,
+    Api, Client, CustomResourceExt, ResourceExt,
+};
+use openapi::{apis::StatusCode, clients};
+use tracing::{error, info};
 
 /// Get the DiskPool api.
-pub(crate) fn api(client: &Client, namespace: &str) -> Api<DiskPool> {
+pub(crate) fn v1beta1_api(client: &Client, namespace: &str) -> Api<NewDiskPool> {
     Api::namespaced(client.clone(), namespace)
 }
 
 /// Create a disk pool CR, with the given name and spec.
```

Comment:
`/// Create a disk pool CR, with the given name and spec.` --> `/// Create a v1beta1 disk pool CR, with the given name and spec.`